File: lib_base/src/main/java/com/niuda/a3jidi/lib_base/base/http/HttpException.java
package com.niuda.a3jidi.lib_base.base.http;

public class HttpException extends RuntimeException {

    /**
	 * 序列化
	 */
	private static final long serialVersionUID = 1L;
	public static final int USER_NOT_EXIST = 100;
    public static final int WRONG_PASSWORD = 101;

    public HttpException(int resultCode) {
        this(getExceptionMessage(resultCode));
    }

    public HttpException(String detailMessage) {
        super(detailMessage);
    }

    /**
     * 由于服务器传递过来的错误信息直接给用户看的话，用户未必能够理解
     * 需要根据错误码对错误信息进行一个转换，在显示给用户
     * @param code
     * @return
     */
    private static String getExceptionMessage(int code){
        String message = "";
        switch (code) {
            case USER_NOT_EXIST:
                message = "该用户不存在";
                break;
            case WRONG_PASSWORD:
                message = "密码错误";
                break;
            default:
                message = "未知错误";

        }
        return message;
    }
}

File: lib_base/src/main/java/com/niuda/a3jidi/lib_base/base/http/HttpException.kt
package com.niuda.a3jidi.lib_base.base.http

class HttpException(detailMessage: String) : RuntimeException(detailMessage) {

    constructor(resultCode: Int) : this(getExceptionMessage(resultCode)) {}

    companion object {

        /**
         * 序列化
         */
        private val serialVersionUID = 1L
        val USER_NOT_EXIST = 100
        val WRONG_PASSWORD = 101

        /**
         * 由于服务器传递过来的错误信息直接给用户看的话，用户未必能够理解
         * 需要根据错误码对错误信息进行一个转换，在显示给用户
         * @param code
         * @return
         */
        private fun getExceptionMessage(code: Int): String {
            var message = ""
            when (code) {
                USER_NOT_EXIST -> message = "该用户不存在"
                WRONG_PASSWORD -> message = "密码错误"
                else -> message = "未知错误"
            }
            return message
        }
    }
}

